File: FarmProject.Domain/Models/Order.cs
using FarmProject.Domain.Common;
using FarmProject.Domain.Constants;

namespace FarmProject.Domain.Models;

public class Order(
        int customerId,
        DateTime orderDate) 
    : Entity
{
    public int CustomerId { get; private set; } = customerId;
    public OrderStatus OrderStatus { get; private set; } = OrderStatus.Processing;
    public DateTime OrderDate { get; private set; } = orderDate;
    public List<OrderRequest> OrderRequests { get; private set; } = new();

    public Result UpdateStatusBasedOnOrderRequests()
    {
        if (OrderRequests.All(or => or.OrderRequestStatus == OrderRequestStatus.Completed))
            OrderStatus = OrderStatus.Completed;
        else if (OrderRequests.Any(or => or.OrderRequestStatus == OrderRequestStatus.Failed))
            OrderStatus = OrderStatus.Failed;
        else
            OrderStatus = OrderStatus.Processing;

        return Result.Success();
    }
}

File: FarmProject.React/src/api/types/Order.ts
export interface OrderRequest {
  id: number;
  orderId: number;
  offspringType: number;
  amount: number;
  sacrificedAmount: number;
  cageId: number;
  orderRequestStatus: number;
}

export interface OrderDetails {
  id: number;
  customerId: number;
  orderStatus: number;
  orderDate: string;
  orderRequests: OrderRequest[];
}

// Replace enum with const object and type
export const OrderRequestStatus = {
  Waiting: 0,
  Completed: 1,
  Failed: 2,
} as const;

export type OrderRequestStatusType = typeof OrderRequestStatus[keyof typeof OrderRequestStatus];

export const orderRequestStatusLabels: Record<OrderRequestStatusType, string> = {
  [OrderRequestStatus.Waiting]: "Waiting",
  [OrderRequestStatus.Completed]: "Completed",
  [OrderRequestStatus.Failed]: "Failed",
};

export const getOrderRequestStatusLabel = (status: number): string => {
  return orderRequestStatusLabels[status as OrderRequestStatusType] || "Unknown";
};

export const getOrderRequestStatusColor = (
  status: number
):
  | "default"
  | "primary"
  | "secondary"
  | "error"
  | "info"
  | "success"
  | "warning" => {
  switch (status) {
    case OrderRequestStatus.Waiting:
      return "warning";
    case OrderRequestStatus.Completed:
      return "success";
    case OrderRequestStatus.Failed:
      return "error";
    default:
      return "default";
  }
};